pkg/agent/multicast/multicast_route.go

```diff
@@ -0,0 +1,422 @@
+}
+
+func (c *MRouteClient) deleteInboundMrouteEntryByGroup(group net.IP) (err error) {
+	klog.V(2).InfoS("Deleting multicast group", "group", group)
+	mEntries, _ := c.inboundRouteCache.ByIndex(GroupNameIndexName, group.String())
+	for _, route := range mEntries {
+		entry := route.(*inboundMulticastRouteEntry)
+		err := c.delMrouteEntry(net.ParseIP(entry.src), net.ParseIP(entry.group), entry.vif)
+		if err != nil {
+			return err
+		}
+		c.inboundRouteCache.Delete(route)
+	}
+	return nil
+}
+
+// addOutBoundMrouteEntry configures multicast route from Antrea gateway to oifs(outbound multicast interfaces)
+// to allow multicast senders in the pods send multicast traffic to external.
+func (c *MRouteClient) addOutBoundMrouteEntry(src net.IP, group net.IP, oifs []string) (err error) {
+	klog.InfoS("Adding outbound multicast route", "src", src, "group", group, "oifs", oifs)
+	vifs := make([]uint16, 0, len(oifs))
+	for _, oif := range oifs {
+		oifvif, err := c.vifAllocator.setVifIfNotExist(oif)
+		if err != nil {
+			return err
+		}
+		vifs = append(vifs, oifvif)
+	}
+
+	inboundVif, _ := c.vifAllocator.getVif(c.gatewayInterface)
+	err = c.addMrouteEntry(src, group, uint16(inboundVif), vifs)
+	if err != nil {
+		return err
+	}
+	routeEntry := &outboundMulticastRouteEntry{
+		group: group.String(),
+		src:   src.String(),
+		vifs:  vifs,
+	}
+	_, exist, _ := c.outboundRouteCache.GetByKey(routeEntry.group + "/" + routeEntry.src)
+	if !exist {
+		c.outboundRouteCache.Add(routeEntry)
+	} else {
+		c.outboundRouteCache.Update(routeEntry)
+	}
+	return nil
+}
+
+// addInboundMrouteEntry configures multicast route from external interface to Antrea gateway
+// to allow multicast receivers in the pods receive multicast traffic from external.
+func (c *MRouteClient) addInboundMrouteEntry(src net.IP, group net.IP, inboundVif uint16) (err error) {
+	klog.InfoS("Adding inbound multicast route", "src", src, "group", group, "inboundVif", inboundVif)
+	oif, err := c.vifAllocator.getVif(c.gatewayInterface)
+	if err != nil {
+		return err
+	}
+
+	err = c.addMrouteEntry(src, group, inboundVif, []uint16{oif})
+	if err != nil {
+		return err
+	}
+	routeEntry := &inboundMulticastRouteEntry{
+		group: group.String(),
+		src:   src.String(),
+		vif:   inboundVif,
+	}
+	c.inboundRouteCache.Add(routeEntry)
+	return nil
+}
+
+// inboundMulticastRouteEntry encodes the inbound multicast routing entry.
+// e.g. (10.0.0.55,226.94.9.9)           Iif: wlan0      Oifs: antrea-gw0
+// The oif is always gatewayInterface so we do not put it in the struct.
+type inboundMulticastRouteEntry struct {
+	group string
+	src   string
+	vif   uint16
+}
+
+// outboundMulticastRouteEntry encodes the outbound multicast routing entry.
+// e.g. (10.244.1.4,224.1.2.10)          Iif: antrea-gw0 Oifs: wlo1 wlp3s0 zt5u4zjljr
+// The oif is always gatewayInterface so we do not put it in the struct.
```

Comment:
ditto
And it seems the comment is contradictory

Reply:
updated.